expression/builtin.go

```diff
@@ -244,6 +253,17 @@ type baseBuiltinCastFunc struct {
 	inUnion bool
 }
 
+// implicitArgs returns the implicit arguments of cast functions
+func (b *baseBuiltinCastFunc) implicitArgs() []types.Datum {
```

Comment:
Why use `Datum`? Can we change to another data structure?

Reply:
I think `Datum` is the best structure which is used to exchange implicit arguments.

1. Codec is easily on both sides.
2. The encoded bytes are small.
3. General used in TiDB/TiKV (some implicit arguments do not need any transform)